mcm/json_layer/request.py

```diff
@@ -1864,6 +1866,81 @@ def test_failure(self, message, what='Submission', rewind=False, with_notificati
 
         self.reload()
 
+    def _filter_workflows(self, all_reqmgr_name_list):
+        """
+        Filter a workflow list to pick only those submitted
+        after the latest request injection from McM 
+        based on the McM request history.
+
+        Args:
+            all_reqmgr_name_list (list[str]): List with workflow names.
+                For instance: pdmvserv_task_EXO-RunIISummer15GS-17992__v1_T_231206_053736_1415
+
+        Returns:
+            A subset of workflow names created after the latest request injection.
+        """
+        prepid = self.get_attribute("prepid")
+        history = self.get_attribute("history") or []
+
+        # Filter only those history records related to
+        # submitted step
+        submitted = []
+        for record in history:
+            action = record.get("action")
+            step = record.get("step")
+
+            submitted_record = action == "set status" and step == "submitted"
+            if submitted_record:
+                submitted.append(record)
+
+        if not submitted:
+            self.logger.error(
+                "Unable to find submitted history records for the request: %s. Returning original list",
+                prepid
+            )
+            return all_reqmgr_name_list
+
+        # Parse the date from the McM history record
+        # as UTC.
+        def _parse_mcm_datetime(dt_str):
+            as_datetime = datetime.datetime.strptime(dt_str, "%Y-%m-%d-%H-%M")
+            tz_included = pytz.timezone("Europe/Zurich").localize(as_datetime)
+            as_utc = tz_included.astimezone(pytz.timezone("UTC"))
```

Comment:
This conversion is ambiguous once a year when switching from summer to winter time (or the other way around). What happens then?

Reply:
The date included in that field is computed using the server clock, so when the transition happens it should have been handled automatically. Related to the conversion, I also expect the library to handle this.